public/video-ui/src/components/ScheduledLaunch/ScheduledLaunch.js

```diff
@@ -11,89 +21,310 @@ export default class ScheduledLaunch extends React.Component {
   };
 
   state = {
-    selectedDate: null,
+    selectedScheduleDate: null,
+    selectedEmbargoDate: null,
     showDatePicker: false,
-    showScheduleButton: true
+    propertyName: null,
+    showScheduleButton: true,
+    showScheduleOptions: false,
+    invalidDateError: null
+  };
+
+  componentWillReceiveProps(nextProps) {
+    const embargo =
+      nextProps.video.contentChangeDetails.embargo &&
+      nextProps.video.contentChangeDetails.embargo.date;
+    const scheduledLaunch =
+      nextProps.video.contentChangeDetails.scheduledLaunch &&
+      nextProps.video.contentChangeDetails.scheduledLaunch.date;
+    this.setState({ selectedScheduleDate: scheduledLaunch || embargo });
+    this.setState({ selectedEmbargoDate: embargo || scheduledLaunch });
+  }
+
+  validateDate = (date, propertyName) => {
+    if (!date) {
+      this.setState({ invalidDateError: null });
+      return;
+    }
+
+    const { video: { contentChangeDetails } } = this.props;
+    const scheduledLaunch =
+      contentChangeDetails &&
+      contentChangeDetails.scheduledLaunch &&
+      contentChangeDetails.scheduledLaunch.date;
+    const embargo =
+      contentChangeDetails &&
+      contentChangeDetails.embargo &&
+      contentChangeDetails.embargo.date;
+
+    this.setState({ invalidDateError: null });
```

Comment:
It feels  a bit strange that we are setting this to null here and then potentially setting it to something else below.

Reply:
I think this was to make sure that the alerts from previous failed validations would be cleared after closing the dropdown and selecting a new option.

Reply:
Ok, that makes sense. You could put the whole logic setting the `invalidDateError` inside an if-else with the else clause setting `invalidDateError` to null